src/ops/attention.cc

```diff
@@ -269,6 +263,12 @@ MultiHeadAttention::MultiHeadAttention(FFModel &model,
       qoSeqLength(_query->dims[1].size), kvSeqLength(_key->dims[1].size)
 // bias_initializer(_bias_initializer)
 {
+  // assert shape is valid
+  assert (_query->check_valid());
```

Comment:
Also I wonder if it's possible to just put the `check_valid` checks up in `Operator` as they'll kinda be used in every operator to some degree